src/internal_events/kubernetes_logs.rs

```diff
@@ -0,0 +1,31 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct KubernetesLogsEventReceived<'a> {
+    pub file: &'a str,
+    pub byte_size: usize,
+}
+
+impl InternalEvent for KubernetesLogsEventReceived<'_> {
+    fn emit_logs(&self) {
+        trace!(
+            message = "received one event.",
+            %self.file,
+            rate_limit_secs = 10
```

Comment:
I feel like if someone enabled trace logging they're going to want to see every event.

Reply:
This code is based on the file source events.
I agree though - doesn’t make much sense to rate limit trace level logging.

Reply:
This code is based on the file source events.
I agree though - doesn’t make much sense to rate limit trace level logging.

Reply:
I'm curious if we should disable rate-limiting for trace logs at a lower level?

Reply:
If we do, we should trigger a compiler error if somebody uses `rate_limit_secs` in the code, to prevent the situation where people request rate limiting, but it ends up ignored. That'd be tricky, I'm not sure how to implement it.
I think it's easier to take care of that on a case-by-case level, and just form a general consensus on how to use logging (i.e. single style of message formats, formatting etc). Ideally, we'd just auto-lint the code to ensure it conforms. One of those cases where linter is a better fit.

Reply:
https://rust-lang.github.io/rfcs/0089-loadable-lints.html